pkg/api/types.go

```diff
@@ -121,3 +121,26 @@ type InstallResult struct {
 	// Error describes last error encountered during install operation
 	Error error
 }
+
+// SourceInfo stores information about the source code
+type SourceInfo struct {
+	// Store the current GIT ref which might be a branch or commit id
+	// This information is available in the image as 'io.openshift.build.commit.ref' LABEL
+	Ref string
+
+	// The GIT commit id
+	// This information is available in the image as 'io.openshift.build.commit.id' LABEL
+	ID string
+
+	// Location from where the Source was downloaded
+	// This information is available in the image as 'io.openshift.build.source' LABEL
+	Location string
+
+	// The date of the commit
+	// This information is available in the image as 'io.openshift.build.commit.date' LABEL
+	Date string
+
+	// Author of the commit
+	// This information is available in the image as 'io.openshift.build.commit.author' LABEL
+	Author string
```

Comment:
@bparees the commit message can be pretty long... are you sure you want to have it as label? I mean if you have source location and GIT ref, then you can just fetch the commit and get the message... Alternatively we can truncate the message to some reasonable length, or just take the first line.